addons/mail/wizard/mail_compose_message.py

```diff
@@ -494,7 +496,7 @@ def _process_state(self, mail_values_dict):
             elif optout_emails and mail_to in optout_emails:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_optout'
-            elif done_emails and mail_to in done_emails:
+            elif done_emails and mail_to in done_emails and not mailing_document_based:
```

Comment:
Shouldn't it depend on `record_id` when it is `mailing_document_based`?
Actually I don't understand why we need the context key if the record is different? (maybe to ensure that we are not breaking anything in stable?)

Reply:
Note that we already loop on res_ids, meaning normally you will never have twice the same document in the loop.

Concerning the context key: because in mass mailing application, you don't want to mail a given email address several times (because mailing lists could contain thousands of emails with duplicated). But when sending invoices in batch, you want to email the same address several times if it has several invoices.

Basically we have two behavior: marketing (don't spam) and business (allow spam).